#! /bin/sh /usr/share/dpatch/dpatch-run
## 02cachedesync.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix for (upstream)#497243: remove causes cache desynchronization
## DP: backed out the attempted fix, as it only dealt with the symptomps 
## DP: and not the root-cause (which is the ill-advised change-set 616)

@DPATCH@
diff -urNad duplicity-0.6.08b~/src/collections.py duplicity-0.6.08b/src/collections.py
--- duplicity-0.6.08b~/src/collections.py	2010-03-12 11:39:06.000000000 +1000
+++ duplicity-0.6.08b/src/collections.py	2010-03-15 20:41:30.904266406 +1000
@@ -145,12 +145,7 @@
             if (pr
                 and pr.time == self.time
                 and pr.start_time == self.start_time
-                and pr.end_time == self.end_time
-                and pr.type != "new-sig" ):
-                # do not remove new sigs from the cache:
-                # they aren't removed from the remote archive,
-                # and subsequent backups will have to resync
-                # which is bad if running non-interactive with encrypt-key 
+                and pr.end_time == self.end_time):
                 try:
                     globals.archive_dir.append(lfn).delete()
                 except:
